src/api/hooks/useGetAPTPrice.ts

```diff
@@ -0,0 +1,30 @@
+const COINGECKO_API_ENDPOINT = "https://api.coingecko.com/api/v3/simple/price";
+
+export async function getAPTPrice(): Promise<number> {
```

Comment:
Thank you for contributing! Some food for thought!

This lives in the `hooks` folder, however this as implemented is not a hook. I like the idea you were going with of having a separate fetch function and hook (ie: `getPrice` and `useGetPrice`). Perhaps export both functions, and the caller can choose which function to use. 

Bellow is a simple example of how you might approach this. Note, this repo uses `react-query` which helps manage caching and state. Your code in `BalanceCard.tsx` can be simplified o remove the use of `useState` if you use `react-query` 

```tsx
import { useQuery } from "@tanstack/react-query";

export function useGetPrice(id: string) {
  return useQuery({
    queryKey: ["price", id],
    queryFn: () => getAPTPrice(id),
  });
}

// No need for state!
const {data: price} = useGetPrice('coin_id')
```

Reply:
Understood. I've implemented the useGetPrice() hook using react query. Let me know if I've done it right and if any more changes need to made. I've also kept the getPrice() function there as it is. Would it make sense to add the getPrice() function to api/index.ts and keep just the hook in the useGetPrice.ts file?